api/v1alpha1/hardwareclassification_types.go

```diff
@@ -27,111 +27,135 @@ type HardwareClassificationSpec struct {
 }
 
 // Disk contains disk details extracted from the hardware profile
 type Disk struct {
 	// +optional
 	// +kubebuilder:validation:Minimum=1
-	// MinimumCount of disk should be greater than 0
-	// MinimumCount > 0
-	MinimumCount int `json:"minimumCount,omitempty"`
+	MinimumCount int `json:"minimumCount" description:"minimum count of disk, greater than 0"`
 	// +optional
 	// +kubebuilder:validation:Minimum=1
-	// MinimumIndividualSizeGB should be greater than 0
-	// Ex. MinimumIndividualSizeGB > 0
-	MinimumIndividualSizeGB int64 `json:"minimumIndividualSizeGB,omitempty"`
+	MinimumIndividualSizeGB int64 `json:"minimumIndividualSizeGB" description:"minimum individual size of disk, greater than 0"`
 	// +optional
 	// +kubebuilder:validation:Minimum=1
-	// MaximumCount of disk should be greater than 0 and greter than MinimumCount
-	// Ex. MaximumCount > 0 && MaximumCount > MinimumCount
-	MaximumCount int `json:"maximumCount,omitempty"`
+	MaximumCount int `json:"maximumCount" description:"maximum count of disk, greater than 0"`
 	// +optional
 	// +kubebuilder:validation:Minimum=1
-	// Maximum individual size should be greater than 0 and greater than MinimumIndividualSizeGB
-	// Ex. MaximumIndividualSizeGB > 0 && MaximumIndividualSizeGB > MinimumIndividualSizeGB
-	MaximumIndividualSizeGB int64 `json:"maximumIndividualSizeGB,omitempty"`
+	MaximumIndividualSizeGB int64 `json:"maximumIndividualSizeGB" description:"maximum individual size of disk, greater than 0"`
 }
 
-// Nic contains nic details extracted from the hardware profile
-type Nic struct {
+// NIC contains nic details extracted from the hardware profile
+type NIC struct {
 	// +optional
 	// +kubebuilder:validation:Minimum=1
-	// Minimum count should be greater than 0
-	// Ex. MinumumCount > 0
-	MinimumCount int `json:"minimumCount,omitempty"`
+	MinimumCount int `json:"minimumCount" description:"minimum count of nics, greater than 0"`
 	// +optional
 	// +kubebuilder:validation:Minimum=1
-	// Maximum count should be greater than 0 and grear than MinimumCount
-	// Ex. MaximumCount > 0 && MaximumCount > MinimumCount
-	MaximumCount int `json:"maximumCount,omitempty"`
+	MaximumCount int `json:"maximumCount" description:"maximum count of nics, greater than 0"`
 }
 
-// Ram contains ram details extracted from the hardware profile
-type Ram struct {
+// RAM contains ram details extracted from the hardware profile
+type RAM struct {
 	// +optional
 	// +kubebuilder:validation:Minimum=1
-	// MinimumSizeGB of Ram should be greater than 0
-	// Ex. MinimumSizeGB > 0
-	MinimumSizeGB int `json:"minimumSizeGB,omitempty"`
+	MinimumSizeGB int `json:"minimumSizeGB" description:"minimun size of ram, greater than 0"`
 	// +optional
 	// +kubebuilder:validation:Minimum=1
-	// MaximumSizeGB should be greater than 0 or greater than MinimumSizeGB
-	// Ex. MaximumSizeGB > 0 && MaximumSizeGB > MinimumSizeGB
-	MaximumSizeGB int `json:"maximumSizeGB,omitempty"`
```

Comment:
I am really worried now. I just checked, your forks are not updated with upstream/master. I want you to fix that first because this PR is removing all the merged code which will cause lot of problems which is addressed in earlier PR's

Reply:
Resolved updated with merged changes.